lisa/main.py

```diff
@@ -87,9 +87,17 @@ def initialize_runtime_folder(
     working_path: Optional[Path] = None,
     run_id: str = "",
 ) -> None:
-    cache_path = _runtime_root.joinpath("cache")
-    cache_path.mkdir(parents=True, exist_ok=True)
-    constants.CACHE_PATH = cache_path
+    # If working_path is provided and absolute, set cache_path within it
```

Comment:
Please just move the original code after the working_path is set. It doesn't need to use the _runtime_root, because it's used by working_path when working_path is set.